openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -59,6 +60,62 @@ const Carousel = {
             slidesToScroll: a,
             responsive: responsive_settings
         });
+
+        var availabilityStatuses = {
+            'open': {cls: 'cta-btn--available', cta: 'Read'},
+            'borrow_available': {cls: 'cta-btn--available', cta: 'Borrow'},
+            'borrow_unavailable': {cls: 'cta-btn--unavailable', cta: 'Join Waitlist'},
+            'error': {cls: 'cta-btn--missing', cta: 'No eBook'}
+        };
+
+        var addWork = function(work) {
+            var availability = work.availability.status;
+            var ocaid = work.availability.identifier;
+            var cls = availabilityStatuses[availability].cls;
+            var url = (cls == 'cta-btn--available') ?
+                ('/borrow/ia/' + ocaid) : (cls == 'cta-btn--unavailable') ?
+                    ('/books/' + work.availability.openlibrary_edition) : work.key;
+            var cta = availabilityStatuses[availability].cta;
+            var isClickable = availability == 'error' ? 'disabled' : '';
+
+            return '<div class="book carousel__item slick-slide slick-active" ' +
+                '"aria-hidden="false" role="option">' +
+                '<div class="book-cover">' +
+                  '<a href="' + work.key + '" ' + isClickable + '>' +
+                    '<img class="bookcover" width="130" height="200" title="' + work.title + '" ' +
+                         'src="//covers.openlibrary.org/b/id/' + work.cover_id + '-M.jpg">' +
+                  '</a>' +
+                '</div>' +
+                '<div class="book-cta">' +
+                  '<a class="btn cta-btn ' + cls + '" href="' + url + '" data-ol-link-track="subjects" ' +
+                  'title="' + cta + ': ' + work.title + '" data-key="subjects" data-ocaid="' + ocaid + '">' + cta + '</a>' +
+                '</div>' +
+              '</div>';
```

Comment:
I'm in pseudocode land right now due to time constraints, but I wonder if we can move this block of HTML to either a `<template>` element or a `<script type="template">` so it's something like:

```
'<div class="book carousel__item slick-slide slick-active" "aria-hidden="false" role="option"><div class="book-cover"><a href="{key}" class="{isClickable}"><img class="bookcover" width="130" height="200" title="{title}" src="//covers.openlibrary.org/b/id/{cover_id}-M.jpg"></a></div><div class="book-cta"><a class="btn cta-btn {cls}" href="{url}" data-ol-link-track="subjects" title="{cta}: {title}" data-key="subjects" data-ocaid="{ocaid}">{cta}</a></div></div>';
```

and then we can write this:

```
var availability = work.availability.status;
var props = {
  key: work.key,
  title: work.title,
  ocaid: work.availability.identifier,
  cover_id: work.cover_id,
  cls: availabilityStatuses[availability].cls,
  url: (cls == 'cta-btn--available') ? ('/borrow/ia/' + ocaid) : 
    (cls == 'cta-btn--unavailable') ? ('/books/' + work.availability.openlibrary_edition) :
    work.key,
  cta: availabilityStatuses[availability].cta,
  isClickable: availability == 'error' ? 'disabled' : ''
};

return tmpl.replace(/\{(\w+)\}/g, function(s, c) {
  return props[c];
});
```

which may get you back under the size constraints and keep the markup where it belongs.